tools/retry

```diff
@@ -16,15 +17,27 @@ if [ "$RETRY" = "0" ]; then
     run_once "$@"
 else
     n=1
+    FAILURE_COUNTER=0
     while :; do
         if [ "$STABILITY_TEST" = "0" ]; then
             [ $n -le "$RETRY" ] || exit 1
             [ $n -eq 0 ] || echo "Retry $n of $RETRY …"
             run_once "$@" && break
         else
             [ $n -le "$RETRY" ] || exit 0
-            run_once "$@" || exit
-            echo "Rerun $n of $RETRY …"
+            if ! run_once "$@"; then
+              if [ "$STABILITY_ALLOW_FAILURE" = "1" ]; then
+                FAILURE_COUNTER=$((FAILURE_COUNTER + 1))
+              else 
+                exit 1
+              fi
+            fi
+            MESSAGE="Rerun $n of $RETRY …"
+            if [ "$STABILITY_ALLOW_FAILURE" = "1" ]; then
+              FAILURE_PERCENTAGE=$(( ( $FAILURE_COUNTER * 100 ) / $n ))
+              MESSAGE= "$MESSAGE with failures: $FAILURE_COUNTER ($FAILURE_PERCENTAGE%)"
```

Comment:
Instead of re-implementing https://github.com/okurz/retry/blob/main/count-fail-ratio you can also just use it, e.g. either install it or download dynamically with "curl"

Reply:
How would I remember allo of your side-projects? :) also I think is better to not overengineer and add those 13 lines to the retry, but you are the po so the choise is yours!

Reply:
I've splitted the change in the script in another pr